src/AI/agents/agents/services/git/git_ops.py

```
@@ -0,0 +1,687 @@
+                        # Use regex replacement for pattern-based
+                        new_content = re.sub(pattern, text, content)
+                        
+                        if new_content == content:
+                            print(f"Warning: Pattern '{pattern}' did not match anything in {file_path}")
+                            continue
+                        
+                        with open(file_path, 'w', encoding='utf-8') as f:
+                            f.write(new_content)
+                        
+                        print(f"  Replaced text using pattern in {file_path}")
+                        modified_files.append(str(file_path))
```

Ruff (0.14.8)

Undefined name `modified_files`

(F821)